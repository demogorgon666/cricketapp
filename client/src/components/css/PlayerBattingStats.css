@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

.bat-main {
  display: flex;
  flex-direction: row;
  margin-top: 210px;
  margin-bottom: 40px;
  justify-content: center;
  width: 95vw;
}

.battting-main-div {
  display: grid;
  grid-template-columns: 39vw 39vw;
}

#batting-format-heading {
  background-color: #2980b9;
  color: white;
  padding: 8px;
  margin-top: 0px;
}

.battting-Test-div {
  border: 1px solid #707070;
  border-top: none;
  margin-right: 10px;
  font-family: "Montserrat", sans-serif;
}

.battting-Odi-div {
  border: 1px solid #707070;
  border-top: none;
  margin-left: 10px;
}

.battting-T20-div {
  border: 1px solid #707070;
  border-top: none;
  margin-right: 10px;
  margin-top: 25px;
  margin-bottom: 25px;
}

.battting-Ipl-div {
  border: 1px solid #707070;
  border-top: none;
  margin-left: 10px;
  margin-top: 25px;
  margin-bottom: 25px;
}

.batting-stats-info-parent-div {
  display: flex;
  flex-direction: row;
  margin-left: 16px;
  margin-right: 16px;
  border-bottom: 1px solid #707070;
  padding-bottom: 16px;
  padding-top: 10px;
  font-family: "Montserrat", sans-serif;
}

.batting-stats-info-div {
  display: flex;
  flex-direction: column;
  text-align: center;
  font-family: "Montserrat", sans-serif;
  padding-left: 20px;
}

#bat-stat-span-1 {
  font-weight: 500;
  font-size: 19px;
}

#bat-stat-span-2 {
  font-weight: 400;
  font-size: 13px;
  padding-top: 2px;
}

@media only screen and (max-width: 576px) {
  .bat-main {
    display: flex;
    flex-direction: row;
    margin-top: 230px;
    justify-content: center;
    width: 100vw;
  }

  .battting-main-div {
    display: grid;
    grid-template-columns: 87vw;
  }

  #batting-format-heading {
    background-color: #2980b9;
    color: white;
    padding: 4px;
    margin-top: 0px;
    font-size: 16px;
  }

  .battting-Test-div {
    border: 1px solid #707070;
    border-top: none;
    margin: auto;
    margin-top: 15px;
    font-family: "Montserrat", sans-serif;
  }

  .battting-Odi-div {
    border: 1px solid #707070;
    border-top: none;
    margin: auto;
    margin-top: 15px;
  }

  .battting-T20-div {
    border: 1px solid #707070;
    border-top: none;
    margin: auto;
    margin-top: 15px;
  }

  .battting-Ipl-div {
    border: 1px solid #707070;
    border-top: none;
    margin: auto;
    margin-top: 15px;
  }

  .batting-stats-info-parent-div {
    display: flex;
    flex-direction: row;
    margin-left: 10px;
    margin-right: 10px;
    border-bottom: 1px solid #707070;
    padding-bottom: 3px;
    padding-top: 3px;
    font-family: "Montserrat", sans-serif;
  }

  .batting-stats-info-div {
    display: flex;
    flex-direction: column;
    text-align: center;
    font-family: "Montserrat", sans-serif;
    padding-left: 14px;
  }

  #bat-stat-span-1 {
    font-weight: 500;
    font-size: 13px;
  }

  #bat-stat-span-2 {
    font-weight: 400;
    font-size: 8px;
    padding-top: 3px;
  }
}
