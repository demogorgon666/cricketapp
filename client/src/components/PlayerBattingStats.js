import React, { Component } from "react";
import "../components/css/PlayerBattingStats.css";
import {
  getODIBatsmanStats,
  getTestBatsmanStats,
  getT20BatsmanStats,
  getBatsmanStats
} from "../actions/PlayerAction";
import { connect } from "react-redux";

export class PlayerBattingStats extends Component {
  componentDidMount() {
    console.log("from mount", this.props.player_id);
    this.props.getODIBatsmanStats(this.props.player_id);

    this.props.getT20BatsmanStats(this.props.player_id);
    this.props.getTestBatsmanStats(this.props.player_id);
  }

  render() {
    console.log(
      "Component batting stats ",
      this.props.t20BatsmanStats["0"].TotalRuns
        ? this.props.t20BatsmanStats["0"].TotalRuns.length === 0
          ? console.log("not returned")
          : this.props.t20BatsmanStats["0"].TotalRuns[0].match_type
        : console.log("not found")
    );

    return (
      <div className="bat-main">
        <div className="battting-main-div">
          <div className="battting-Test-div">
            <h3 id="batting-format-heading" className="jest-Heading-1">
              Test
            </h3>
            <div className="batting-stats-info-parent-div">
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].Matches
                    ? this.props.batsmanStats["0"].Matches.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].Matches[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Matches</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">623</span>
                <span id="bat-stat-span-2">Innings</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].TotalRuns
                    ? this.props.batsmanStats["0"].TotalRuns.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].TotalRuns[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Runs</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].BallsFaced
                    ? this.props.batsmanStats["0"].BallsFaced.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].BallsFaced[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Balls Faced</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].StrikeRate
                    ? this.props.batsmanStats["0"].StrikeRate.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].StrikeRate[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Strike Rate</span>
              </div>
              {/* <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">43</span>
                <span id="bat-stat-span-2">Average</span>
              </div> */}
            </div>
            <div
              className="batting-stats-info-parent-div"
              style={{ borderBottom: "none" }}
            >
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].Highestscore
                    ? this.props.batsmanStats["0"].Highestscore.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].Highestscore[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Highest Score</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].NotOut
                    ? this.props.batsmanStats["0"].NotOut.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].NotOut[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Not Out</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].Hundreds
                    ? this.props.batsmanStats["0"].Hundreds.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].Hundreds[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">100s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].Fifties
                    ? this.props.batsmanStats["0"].Fifties.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].Fifties[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">50s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].TwoHundreds
                    ? this.props.batsmanStats["0"].TwoHundreds.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].TwoHundreds[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">200s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].Fours
                    ? this.props.batsmanStats["0"].Fours.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].Fours[0].player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">4s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.batsmanStats["0"].Sixes
                    ? this.props.batsmanStats["0"].Sixes.length === 0
                      ? "-"
                      : this.props.batsmanStats["0"].Sixes[0].player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">6s</span>
              </div>
            </div>
          </div>
          <div className="battting-Odi-div">
            <h3 id="batting-format-heading" className="jest-Heading-3">
              ODI
            </h3>
            <div className="batting-stats-info-parent-div">
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].Matches
                    ? this.props.odiBatsmanStats["0"].Matches.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].Matches[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Matches</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">623</span>
                <span id="bat-stat-span-2">Innings</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].TotalRuns
                    ? this.props.odiBatsmanStats["0"].TotalRuns.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].TotalRuns[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Runs</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].BallsFaced
                    ? this.props.odiBatsmanStats["0"].BallsFaced.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].BallsFaced[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Balls Faced</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].StrikeRate
                    ? this.props.odiBatsmanStats["0"].StrikeRate.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].StrikeRate[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Strike Rate</span>
              </div>
            </div>
            <div
              className="batting-stats-info-parent-div"
              style={{ borderBottom: "none" }}
            >
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].Highestscore
                    ? this.props.odiBatsmanStats["0"].Highestscore.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].Highestscore[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Highest Score</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].NotOut
                    ? this.props.odiBatsmanStats["0"].NotOut.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].NotOut[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Not Out</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].Hundreds
                    ? this.props.odiBatsmanStats["0"].Hundreds.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].Hundreds[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">100s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].Fifties
                    ? this.props.odiBatsmanStats["0"].Fifties.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].Fifties[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">50s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].TwoHundreds
                    ? this.props.odiBatsmanStats["0"].TwoHundreds.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].TwoHundreds[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">200s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].Fours
                    ? this.props.odiBatsmanStats["0"].Fours.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].Fours[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">4s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.odiBatsmanStats["0"].Sixes
                    ? this.props.odiBatsmanStats["0"].Sixes.length === 0
                      ? "-"
                      : this.props.odiBatsmanStats["0"].Sixes[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">6s</span>
              </div>
            </div>
          </div>
          <div className="battting-T20-div">
            <h3 id="batting-format-heading" className="jest-Heading-2">
              T20
            </h3>
            <div className="batting-stats-info-parent-div">
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Matches
                    ? this.props.t20BatsmanStats["0"].Matches.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Matches[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Matches</span>
              </div>

              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Matches
                    ? this.props.t20BatsmanStats["0"].Matches.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Matches[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Runs</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].BallsFaced
                    ? this.props.t20BatsmanStats["0"].BallsFaced.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].BallsFaced[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Balls Faced</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].StrikeRate
                    ? this.props.t20BatsmanStats["0"].StrikeRate.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].StrikeRate[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Strike Rate</span>
              </div>
            </div>
            <div
              className="batting-stats-info-parent-div"
              style={{ borderBottom: "none" }}
            >
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Highestscore
                    ? this.props.t20BatsmanStats["0"].Highestscore.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Highestscore[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Highest Score</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].NotOut
                    ? this.props.t20BatsmanStats["0"].NotOut.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].NotOut[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">Not Out</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Hundreds
                    ? this.props.t20BatsmanStats["0"].Hundreds.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Hundreds[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">100s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Fifties
                    ? this.props.t20BatsmanStats["0"].Fifties.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Fifties[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">50s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].TwoHundreds
                    ? this.props.t20BatsmanStats["0"].TwoHundreds.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].TwoHundreds[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">200s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Fours
                    ? this.props.t20BatsmanStats["0"].Fours.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Fours[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">4s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">
                  {this.props.t20BatsmanStats["0"].Sixes
                    ? this.props.t20BatsmanStats["0"].Sixes.length === 0
                      ? "-"
                      : this.props.t20BatsmanStats["0"].Sixes[0]
                          .player_stats_value
                    : console.log("not found")}
                </span>
                <span id="bat-stat-span-2">6s</span>
              </div>
            </div>
          </div>
          {/* <div className="battting-Ipl-div">
            <h3 id="batting-format-heading">IPL</h3>
            <div className="batting-stats-info-parent-div">
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">500</span>
                <span id="bat-stat-span-2">Matches</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">623</span>
                <span id="bat-stat-span-2">Innings</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">21756</span>
                <span id="bat-stat-span-2">Runs</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">37568</span>
                <span id="bat-stat-span-2">Balls Faced</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">123.4</span>
                <span id="bat-stat-span-2">Strike Rate</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">43</span>
                <span id="bat-stat-span-2">Average</span>
              </div>
            </div>
            <div
              className="batting-stats-info-parent-div"
              style={{ borderBottom: "none" }}
            >
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">375</span>
                <span id="bat-stat-span-2">Highest Score</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">125</span>
                <span id="bat-stat-span-2">Not Out</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">30</span>
                <span id="bat-stat-span-2">100s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">55</span>
                <span id="bat-stat-span-2">50s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">3</span>
                <span id="bat-stat-span-2">200s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">2749</span>
                <span id="bat-stat-span-2">4s</span>
              </div>
              <div className="batting-stats-info-div">
                <span id="bat-stat-span-1">2140</span>
                <span id="bat-stat-span-2">6s</span>
              </div>
            </div>
          </div> */}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  batsmanStats: state.PlayerReducer.batsmanStats,
  odiBatsmanStats: state.PlayerReducer.odiBatsmanStats,
  t20BatsmanStats: state.PlayerReducer.t20BatsmanStats
});

export default connect(
  mapStateToProps,
  {
    getODIBatsmanStats,
    getTestBatsmanStats,
    getT20BatsmanStats,
    getBatsmanStats
  }
)(PlayerBattingStats);
