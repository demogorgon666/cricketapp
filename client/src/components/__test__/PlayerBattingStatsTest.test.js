import React from "react";
import { shallow, mount } from "enzyme";
import { PlayerBattingStats } from "../PlayerBattingStats";
// import { getBatsmanStats } from "../../actions/PlayerAction";

const getBatsmanStats = jest.fn();
const getODIBatsmanStats = jest.fn();
const getT20BatsmanStats = jest.fn();
const getTestBatsmanStats = jest.fn();
const batsmanStats = [[]];
const t20BatsmanStats = [[]];
const odiBatsmanStats = [[]];

const wrapper = shallow(
  <PlayerBattingStats
    getBatsmanStats={getBatsmanStats}
    getODIBatsmanStats={getODIBatsmanStats}
    getT20BatsmanStats={getT20BatsmanStats}
    getTestBatsmanStats={getTestBatsmanStats}
    batsmanStats={batsmanStats}
    t20BatsmanStats={t20BatsmanStats}
    odiBatsmanStats={odiBatsmanStats}
  />
);

describe("Testing Player Batting stats component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should Have 4 headings Test,ODI,T20,IPL", () => {
    expect(
      wrapper
        .find("h3")
        .at(0)
        .text()
    ).toBe("Test");
    expect(
      wrapper
        .find("h3")
        .at(1)
        .text()
    ).toBe("ODI");
    expect(
      wrapper
        .find("h3")
        .at(2)
        .text()
    ).toBe("T20");
  });
  it("should have 70 span tags", () => {
    expect(
      wrapper
        .find("span")
        .at(1)
        .text()
    ).toBe("Matches");
  });
});
