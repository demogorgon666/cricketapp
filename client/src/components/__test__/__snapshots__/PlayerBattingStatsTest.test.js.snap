// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Player Batting stats component should render the component 1`] = `
<div
  className="bat-main"
>
  <div
    className="battting-main-div"
  >
    <div
      className="battting-Test-div"
    >
      <h3
        className="batting-format-heading"
        id="jest-Heading-1"
      >
        Test
      </h3>
      <div
        className="batting-stats-info-parent-div"
      >
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Matches"
          >
            34
          </span>
          <span
            className="bat-stat-span-2"
          >
            Matches
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Innings"
          >
            34
          </span>
          <span
            className="bat-stat-span-2"
          >
            Innings
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-TotalRuns"
          >
            234
          </span>
          <span
            className="bat-stat-span-2"
          >
            Runs
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-BallsFaced"
          />
          <span
            className="bat-stat-span-2"
          >
            Balls Faced
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-StrikeRate"
          />
          <span
            className="bat-stat-span-2"
          >
            Strike Rate
          </span>
        </div>
      </div>
      <div
        className="batting-stats-info-parent-div"
        style={
          Object {
            "borderBottom": "none",
          }
        }
      >
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Highestscore"
          >
            69
          </span>
          <span
            className="bat-stat-span-2"
          >
            Highest Score
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-NotOut"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            Not Out
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Hundreds"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            100s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Fifties"
          />
          <span
            className="bat-stat-span-2"
          >
            50s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-TwoHundreds"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            200s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Fours"
          />
          <span
            className="bat-stat-span-2"
          >
            4s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-Test-Format-Sixes"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            6s
          </span>
        </div>
      </div>
    </div>
    <div
      className="battting-Odi-div"
    >
      <h3
        className="batting-format-heading"
        id="jest-Heading-2"
      >
        ODI
      </h3>
      <div
        className="batting-stats-info-parent-div"
      >
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Matches
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Innings
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Runs
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Balls Faced
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Strike Rate
          </span>
        </div>
      </div>
      <div
        className="batting-stats-info-parent-div"
        style={
          Object {
            "borderBottom": "none",
          }
        }
      >
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Highest Score
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            Not Out
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            100s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            50s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            200s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            4s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
          />
          <span
            className="bat-stat-span-2"
          >
            6s
          </span>
        </div>
      </div>
    </div>
    <div
      className="battting-T20-div"
    >
      <h3
        className="batting-format-heading"
        id="jest-Heading-3"
      >
        T20
      </h3>
      <div
        className="batting-stats-info-parent-div"
      >
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Matches"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            Matches
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Innings"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            Innings
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-TotalRuns"
          >
            234
          </span>
          <span
            className="bat-stat-span-2"
          >
            Runs
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-BallsFaced"
          />
          <span
            className="bat-stat-span-2"
          >
            Balls Faced
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-StrikeRate"
          />
          <span
            className="bat-stat-span-2"
          >
            Strike Rate
          </span>
        </div>
      </div>
      <div
        className="batting-stats-info-parent-div"
        style={
          Object {
            "borderBottom": "none",
          }
        }
      >
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Highestscore"
          />
          <span
            className="bat-stat-span-2"
          >
            Highest Score
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-NotOut"
          >
            3
          </span>
          <span
            className="bat-stat-span-2"
          >
            Not Out
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Hundreds"
          >
            6
          </span>
          <span
            className="bat-stat-span-2"
          >
            100s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Fifties"
          >
            -
          </span>
          <span
            className="bat-stat-span-2"
          >
            50s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-TwoHundreds"
          >
            1
          </span>
          <span
            className="bat-stat-span-2"
          >
            200s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Fours"
          >
            69
          </span>
          <span
            className="bat-stat-span-2"
          >
            4s
          </span>
        </div>
        <div
          className="batting-stats-info-div"
        >
          <span
            className="bat-stat-span-1"
            id="jest-T20-Format-Sixes"
          >
            34
          </span>
          <span
            className="bat-stat-span-2"
          >
            6s
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
